apps/framework-cli/src/framework/controller.rs

```diff
@@ -134,17 +134,18 @@ pub fn get_all_framework_objects(
     framework_objects: &mut HashMap<String, FrameworkObject>,
     schema_dir: &Path,
     version: &str,
+    aggregations: &HashSet<String>,
 ) -> anyhow::Result<()> {
     if schema_dir.is_dir() {
         for entry in std::fs::read_dir(schema_dir)? {
             let entry = entry?;
             let path = entry.path();
             if path.is_dir() {
                 debug!("<DCM> Processing directory: {:?}", path);
-                get_all_framework_objects(framework_objects, &path, version)?;
+                get_all_framework_objects(framework_objects, &path, version, aggregations)?;
             } else if is_schema_file(&path) {
                 debug!("<DCM> Processing file: {:?}", path);
-                let objects = get_framework_objects_from_schema_file(&path, version)?;
+                let objects = get_framework_objects_from_schema_file(&path, version, aggregations)?;
```

Comment:
It should be fine if a data model in an old version have the same name as an aggregation.

Reply:
How do they get to that point? Like, what would their repro steps be over time?

Reply:
create new version
delete data model
create aggregation with the same name

this might be quite implausible, but if it's worth preventing the name clashing of the two types, i think it's equally worth preventing false positives.

Reply:
Agreed, appreciate that mindset. I updated the PR so it disallows the name clashing for current version only. Let me know if there's anything else.